src/core/toga/widgets/icon.py

```diff
@@ -39,7 +39,7 @@ def __init__(self, path, system=False, factory=None):
         else:
             self.filename = self.path
 
-        self._impl = self.factory.Icon(interface=self)
+        self._impl = self.factory.Icon(interface=self, filename=self.filename)
```

Comment:
This doesn't seem right: you should be able to get to the filename by accessing the interface, without having to pass it in explicitly...